    1: import pytest
    2: 
    3: import pandas.core.dtypes.concat as _concat
    4: 
    5: import pandas as pd
    6: from pandas import Series
    7: import pandas._testing as tm
    8: 
    9: 
   10: def test_concat_mismatched_categoricals_with_empty():
   11:     # concat_compat behavior on series._values should match pd.concat on series
   12:     ser1 = Series(["a", "b", "c"], dtype="category")
   13:     ser2 = Series([], dtype="category")
   14: 
   15:     msg = "The behavior of array concatenation with empty entries is deprecated"
   16:     with tm.assert_produces_warning(FutureWarning, match=msg):
   17:         result = _concat.concat_compat([ser1._values, ser2._values])
   18:     with tm.assert_produces_warning(FutureWarning, match=msg):
   19:         expected = pd.concat([ser1, ser2])._values
   20:     tm.assert_categorical_equal(result, expected)
   21: 
   22: 
   23: @pytest.mark.parametrize("copy", [True, False])
   24: def test_concat_single_dataframe_tz_aware(copy):
   25:     # https://github.com/pandas-dev/pandas/issues/25257
   26:     df = pd.DataFrame(
   27:         {"timestamp": [pd.Timestamp("2020-04-08 09:00:00.709949+0000", tz="UTC")]}
   28:     )
   29:     expected = df.copy()
   30:     result = pd.concat([df], copy=copy)
   31:     tm.assert_frame_equal(result, expected)
   32: 
   33: 
   34: def test_concat_periodarray_2d():
   35:     pi = pd.period_range("2016-01-01", periods=36, freq="D")
   36:     arr = pi._data.reshape(6, 6)
   37: 
   38:     result = _concat.concat_compat([arr[:2], arr[2:]], axis=0)
   39:     tm.assert_period_array_equal(result, arr)
   40: 
   41:     result = _concat.concat_compat([arr[:, :2], arr[:, 2:]], axis=1)
   42:     tm.assert_period_array_equal(result, arr)
   43: 
   44:     msg = (
   45:         "all the input array dimensions.* for the concatenation axis must match exactly"
   46:     )
   47:     with pytest.raises(ValueError, match=msg):
   48:         _concat.concat_compat([arr[:, :2], arr[:, 2:]], axis=0)
   49: 
   50:     with pytest.raises(ValueError, match=msg):
   51:         _concat.concat_compat([arr[:2], arr[2:]], axis=1)
